support/cas-server-support-otp-mfa-core/src/main/java/org/apereo/cas/otp/web/flow/OneTimeTokenAccountCheckRegistrationAction.java

```diff
@@ -27,15 +27,25 @@
 
     @Override
     protected Event doExecute(final RequestContext requestContext) {
-        val uid = WebUtils.getAuthentication(requestContext).getPrincipal().getId();
+        val auth = WebUtils.getAuthentication(requestContext);
+        val uid = auth.getPrincipal().getId();
+        val keyUriFmt = "otpauth://totp/" + this.label + ":" + uid + "?secret=%s&issuer=" + this.issuer;
+
+        if (auth.getAttributes().containsKey("isNewOtpRegistration")) {
+            LOGGER.debug("Arrived back at registration check with isNewOtpRegistration set; returning register");
+            return new EventFactorySupport().event(this, "register");
+        }
```

Comment:
This cannot pass. You should not be using authentication attributes as data placeholders. Use the flow itself to pass data between flow states.